src/test/java/it/gov/pagopa/payhub/activities/activity/SendEmailIngestionFlowActivityTest.java

```diff
@@ -0,0 +1,194 @@
+class SendEmailIngestionFlowActivityTest {
+    @Autowired
+    private SendEmailIngestionFlowActivity sendEmailIngestionFlowActivity;
+    @MockBean
+    private IngestionFlowFileDao ingestionFlowFileDao;
+    @MockBean
+    private SendMailService sendMailService;
+    @MockBean
+    private UserAuthorizationService userAuthorizationService;
+
+    private IngestionFlowFileDTO validIngestionFlowFileDTO;
+    private IngestionFlowFileDTO invalidIngestionFlowFileDTO;
+    private UserInfoDTO validUserInfoDTO;
+    private UserInfoDTO invalidUserInfoDTO;
+    private UserInfoDTO validOrganizationInfoDTO;
+
+    @BeforeEach
+    void init() {
+        createBeans();
+    }
+
+    /**
+     * send email for OK loading flow, success: valid user and valid flow
+     */
+    @Test
+    void sendEmailIngestionOkValidUserValidFlowSuccess() throws MessagingException {
+        Long ingestionFlowFileId = 100L;
+        Mockito.when(ingestionFlowFileDao.findById(ingestionFlowFileId)).thenReturn(Optional.of(validIngestionFlowFileDTO));
+        Mockito.when(userAuthorizationService.getUserInfo(validIngestionFlowFileDTO.getOrg().getIpaCode(), validUserInfoDTO.getMappedExternalUserId())).thenReturn(validUserInfoDTO);
+        Mockito.when(userAuthorizationService.getUserInfo(validIngestionFlowFileDTO.getOrg().getIpaCode(),validIngestionFlowFileDTO.getOrg().getIpaCode()+"-WS_USER")).thenReturn(validOrganizationInfoDTO);
+        Mockito.doNothing().when(sendMailService).sendMail(any(MailTo.class));
+
+        Assertions.assertTrue(sendEmailIngestionFlowActivity.sendEmail(ingestionFlowFileId, true));
+    }
+
+    /**
+     * send email for KO loading flow, success: valid user and valid flow
+     */
+    @Test
+    void sendEmailIngestionKoValidUserValidFlowSuccess() throws MessagingException {
+        Long ingestionFlowFileId = 100L;
+        Mockito.when(ingestionFlowFileDao.findById(ingestionFlowFileId)).thenReturn(Optional.of(validIngestionFlowFileDTO));
+        Mockito.when(userAuthorizationService.getUserInfo(validIngestionFlowFileDTO.getOrg().getIpaCode(), validUserInfoDTO.getMappedExternalUserId())).thenReturn(validUserInfoDTO);
+        Mockito.doNothing().when(sendMailService).sendMail(any(MailTo.class));
+
+        Assertions.assertTrue(sendEmailIngestionFlowActivity.sendEmail(ingestionFlowFileId, false));
+    }
+
+
+    /**
+     * send email for OK loading flow, success: valid user and invalid flow
+     */
+    @Test
+    void sendEmailIngestionOkValidUserInvalidFlowFailed() throws MessagingException {
+        Long ingestionFlowFileId = 100L;
+        Mockito.when(ingestionFlowFileDao.findById(ingestionFlowFileId)).thenReturn(Optional.of(invalidIngestionFlowFileDTO));
+        Mockito.when(userAuthorizationService.getUserInfo(invalidIngestionFlowFileDTO.getOrg().getIpaCode(), validUserInfoDTO.getMappedExternalUserId())).thenReturn(validUserInfoDTO);
+        Mockito.doNothing().when(sendMailService).sendMail(any(MailTo.class));
```

Comment:
Instead of any MailTo, could we set an expectedMailTo?
This way we will test its right building logic

Reply:
OK, done